File: apps/router/src/cli/ui/theme/paint.ts
import { colorEnabled } from "./colors";

const TONES = ["accent", "signature", "ok", "warn", "bad", "dim"] as const;
type Tone = (typeof TONES)[number];

const CODE: Record<Tone, number> = { accent: 36, signature: 35, ok: 32, warn: 33, bad: 31, dim: 90 };

export function paint(text: string, tone: Tone, opts?: { bold?: boolean }): string {
  if (!colorEnabled()) return text;
  const seq = (opts?.bold ? "1;" : "") + CODE[tone];
  return `\x1b[${seq}m${text}\x1b[0m`;
}

File: crates/cli/src/paint.rs
#[derive(Clone, Copy)]
pub enum Tone {
    Ok,
    Warn,
    Bad,
}

pub fn color_enabled() -> bool {
    use std::io::IsTerminal;
    std::io::stdout().is_terminal()
        && std::env::var_os("NO_COLOR").is_none()
        && std::env::var("TERM").map(|t| t != "dumb").unwrap_or(true)
}

pub fn paint(text: &str, tone: Tone, bold: bool) -> String {
    if !color_enabled() {
        return text.to_string();
    }
    let code = match tone {
        Tone::Ok => 32,
        Tone::Warn => 33,
        Tone::Bad => 31,
    };
    if bold {
        format!("\x1b[1;{code}m{text}\x1b[0m")
    } else {
        format!("\x1b[{code}m{text}\x1b[0m")
    }
}
